src/store/proposals/proposals.sagas.ts

```diff
@@ -108,10 +115,69 @@ const watchProposalsList = createWatcherSaga({
   stopAction: proposalsActions.stopWatchingProposalsList.type,
 });
 
+export function* addProposal({ payload }: PayloadAction<AddProposalFields>) {
```

Comment:
You can use the `ProposalsActions` type from your `.slice` file

```suggestion
export function* addProposal({ payload }: ProposalsActions['startProposal']) {
```